src/main/java/com/cburch/logisim/gui/start/Startup.java

```diff
@@ -145,388 +155,534 @@ static void doPrint(File file) {
+    String loc = null;
+    var x = 0;
+    var y = 0;
+    if (p >= 0) {
+      loc = wxh[1].substring(p + 1);
+      wxh[1] = wxh[1].substring(0, p);
+      final var xy = loc.split("\\+");
+      if (xy.length != 2 || xy[0].length() < 1 || xy[1].length() < 1) {
+        logger.error(S.get("argGeometryError"));
+        return RC.QUIT;
+      }
+      try {
+        x = Integer.parseInt(xy[0]);
+        y = Integer.parseInt(xy[1]);
+      } catch (NumberFormatException e) {
+        logger.error(S.get("argGeometryError"));
+        return RC.QUIT;
+      }
+    }
+
+    var w = 0;
+    var h = 0;
+    try {
+      w = Integer.parseInt(wxh[0]);
+      h = Integer.parseInt(wxh[1]);
+    } catch (NumberFormatException e) {
+      logger.error(S.get("argGeometryError"));
+      return RC.QUIT;
+    }
+    if (w <= 0 || h <= 0) {
+      logger.error(S.get("argGeometryError"));
+      return RC.QUIT;
+    }
+    AppPreferences.WINDOW_WIDTH.set(w);
+    AppPreferences.WINDOW_HEIGHT.set(h);
+    if (loc != null) {
+      AppPreferences.WINDOW_LOCATION.set(x + "," + y);
+    }
+    return RC.OK;
   }
 
+  private static RC handleArgLocale(Startup startup, Option opt) {
+    setLocale(opt.getValue());
+    return RC.OK;
+  }
+
+  private static RC handleArgAccents(Startup startup, Option opt) {
+    final var flag = opt.getValue().toLowerCase();
+    try {
+      AppPreferences.ACCENTS_REPLACE.setBoolean(!parseBool(flag));
+    } catch (IllegalArgumentException ex) {
+      logger.error(S.get("argAccentsOptionError"));
+      return RC.QUIT;
+    }
+    return RC.OK;
+  }
+
+  private static RC handleArgTemplate(Startup startup, Option opt) {
+    if (startup.templFile != null || startup.templEmpty || startup.templPlain) {
+      logger.error(S.get("argOneTemplateError"));
+      return RC.QUIT;
+    }
+    final var file = opt.getValue();
+    startup.templFile = new File(file);
+    String errMsg = null;
+    if (!startup.templFile.exists()) errMsg = S.get("templateMissingError", file);
+    if (errMsg == null && !startup.templFile.canRead()) errMsg = S.get("templateCannotReadError", file);
+    if (errMsg != null) return RC.WARN;  // FIXME: shouldn't we quit in such case?
+    return RC.OK;
+  }
+
+  private static RC handleArgNoSplash(Startup startup, Option opt) {
+    startup.showSplash = false;
+    return RC.OK;
+  }
+
+  private static RC handleArgTestVector(Startup startup, Option opt) {
+    startup.circuitToTest = opt.getValues()[0];
+    startup.testVector = opt.getValues()[1];
+    startup.showSplash = false;
+    startup.exitAfterStartup = true;
+    // This is to test a test bench. It will return 0 or 1 depending on if the tests pass or not.
+    return RC.OK;
+  }
+
+  private static RC handleArgTestFpgaImpl(Startup startup, Option opt) {
+    final var optArgs = opt.getValues();
+
+    // already handled above
+    startup.testCircuitImpPath = optArgs[0];
+    startup.testCircuitImpMapFile = optArgs[1];
```

Comment:
Mapfile is an optional argument, what happens if it is not passed?

Reply:
Can you then tell me what are possible valid invocations here? Because it may be it's too strict now.

Reply:
```-test-fpga-implementation  <circ_input> [map_input] <circuit_name> <board> [tick frequency] [HDLONLY]```

the ones in [] are optional, the ones in <> are required. [map_input] is for backward-compatibility, and can maybe be dropped. Possible invocations are (without map_input):

```-test-fpga-implementation  <circ_input> <circuit_name> <board> [tick frequency]```

```-test-fpga-implementation  <circ_input> <circuit_name> <board>  [HDLONLY]```

```-test-fpga-implementation  <circ_input> <circuit_name> <board> [tick frequency]  [HDLONLY]```

And than you have the 3 versions with [map_input] (which I think we can drop)

Reply:
Oops, forgot:

```-test-fpga-implementation <circ_input> <circuit_name> <board>```

Reply:
> is for backward-compatibility, and can maybe be dropped.

Can we do that then? It's mess there already. Removal of this will simplify a lot.

Reply:
Yes drop it

Reply:
Also for the tick frequency we should specify if it is only in Hz, or if we allow for:
```4.1k``` or ```1M``` etc.

Reply:
For now it expects valid integer. We can make this smarter w/o problem - please create new ticket for that though

Reply:
Ok, I should have this all covered now. PTAL